### ISSUE TITLE
Evaluate possibility to incorporate changes made to forks

### ISSUE THREAD

**Issue by waldyrious** [hamstar/Wikimate#74]
The changes are relatively straightforward, and some of them may not apply anymore. We should evaluate if that's the case, otherwise we should merge them in.

- [Changes](https://github.com/hamstar/Wikimate/compare/master...devlet:feature/md5) from @devlet's fork -- 2 commits:
    - 9bfa241: Switch methods and variables from private to protected so the class be be easily extended
    - 1144b7c: Avoid MD5 errors
- [Changes](https://github.com/hamstar/Wikimate/compare/master...nasirkhan:master) from @nasirkhan's fork -- 2 commits:
    - be582ab: fixed and updated composer.json 
    - c664f00: updated cURL path
- [Changes](https://github.com/hamstar/Wikimate/compare/master...MhdSyrwan:master) from @MhdSyrwan's fork -- 1 commit:
    - ce1eb62: Moved login() out of the constructor

From a cursory look, these changes seem sensible (if up-to-date), except perhaps for the "avoid MD5 errors" one. Calling the authors to comment here :)

**Comment by waldyrious** [comment]
c664f00 (updated cURL path) probably is obsolete, given the discussion in #21. Can anyone confirm/deny?

**Comment by Xymph** [comment]
Yes, and be582ab was already incorporated: [this](https://github.com/hamstar/Wikimate/commit/10966166fc3180abc1a2667a5eba014d24a34e78#diff-b5d0ee8c97c7abd7e3fa29b9a27d1780) and 9de44eb.

**Comment by waldyrious** [comment]
Awesome, thanks for clarifying. Let's wait for @nasirkhan and @MhdSyrwan\* to comment -- ideally they would sync up their forks (or delete them if they can now use the upstream directly), so that the [network graph](https://github.com/hamstar/Wikimate/network) is cleaned up.

\* @devlet seems to be inactive, so I'm nor sure how realistic it is to expect a reaction from his side.

**Comment by devlet** [comment]
Thanks for incorporating all these changes :)

I could not make `setText()` work without commenting the `md5 => md5($text)` parameter, which I considered ok since it seems redundant as it can be derived directly from the `text` param.

**Comment by Xymph** [comment]
Thanks for popping in here. :) `md5` is a checksum, so while I wouldn't know why it doesn't work in your case, I prefer to keep it enabled by default.
Since the `token` parameter is sent (nearly) at the end of the request ([as recommended](https://www.mediawiki.org/wiki/API:Edit#Parameters)) you should still be guarded against data corruption problems.

**Comment by nasirkhan** [comment]
thanks for looking into the issues. It is great that the project is active again. :smile:

**Comment by devlet** [comment]
It's fine be me if we just close the md5 issue for now.

**Comment by Xymph** [comment]
Well, there isn't an open md5 issue in the main project. waldyrious brought up this topic to see if your fork (and nasirkhan's and MhdSyrwan's too) can be deleted or synced up with the master, in order to clean up the network graph. If so, that takes an action on behalf of all three of you.

**Comment by devlet** [comment]
I tested current master branch with my custom code and it's working without my `md5` change. I just synced my copy of upstream's master branch. Is that enough? Or shall I also delete my `feature/md5` branch to cleanup de graph?

**Comment by waldyrious** [comment]
@devlet if you don't mind, please delete the branches, yes (feature/md5 and feature/protected), as they still appear in the network graph.

**Comment by devlet** [comment]
I deleted both branches :)
